### PR TITLE
chore(tekton): Update to Tekton v0.5.1

### PR DESCRIPTION
**Pull request by abayer**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

Testing Tekton v0.5.0 (waiting for v0.5.1 to release shortly).

#### Which issue this PR fixes

n/a

### CODE REVIEW

**Comment by abayer** [comment]
This is gonna fail - we need the `config-defaults` configmap in the tekton chart.

**Comment by abayer** [comment]
(I manually hacked in the config map for right now)

**Comment by abayer** [comment]
eh, tekton still failed due to `jx get build logs` timing out. Will dig later.

**Comment by hferentschik** [comment]
how is this upgrade going?

**Comment by abayer** [comment]
Poorly. =) I’ll circle back to it once the release process has restabilized.

**Comment by abayer** [comment]
Assuming that the latest commit passes the `tekton` tests, this is good to go - I've verified it works against Tekton 0.5.1, now just to make sure it also works against Tekton 0.4.0 for compatibility reasons. =)

File: pkg/builds/build_info.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func CreateBuildPodInfo(pod *corev1.Pod) *BuildPodInfo {
 	containers, _, isInit := kube.GetContainersWithStatusAndIsInit(pod)
 
 	for _, container := range containers {
-		if strings.HasPrefix(container.Name, "build-step-git-source") {
+		if strings.HasPrefix(container.Name, "build-step-git-source") || strings.HasPrefix(container.Name, "step-git-source") {
```
Comment (dgozalo) [inline pkg/builds/build_info.go]: Wouldn't the second condition also be true for the first one? the first one contains "step-git-source".

File: pkg/logs/tekton_logging.go [hunk 2]
```diff
@@ -46,14 +46,21 @@ func GetTektonPipelinesWithActivePipelineActivity(jxClient versioned.Interface,
...
 	if context != "" {
 		labelsFilter = modifyFilterForPipelineRun(labelsFilter, context)
 	}
 	tektonPRs, _ := tektonClient.TektonV1alpha1().PipelineRuns(ns).List(metav1.ListOptions{
 		LabelSelector: labelsFilter,
 	})
 
+	// Handle the old "repo" tag as well for legacy purposes.
+	if len(tektonPRs.Items) == 0 {
+		labelsFilter = strings.Replace(labelsFilter, "repository=", "repo=", 1)
```
Comment (dgozalo) [inline pkg/logs/tekton_logging.go]: It's going to have a "repository" tag instead of a "repo" one now? Nice.

File: pkg/tekton/pipeline_info.go [hunk 3]
```diff
@@ -176,7 +176,7 @@ func CreatePipelineRunInfo(prName string, podList *corev1.PodList, ps *v1.Pipeli
 	}
 	containers, _, isInit := kube.GetContainersWithStatusAndIsInit(pod)
 	for _, container := range containers {
-		if strings.HasPrefix(container.Name, "build-step-git-source") {
+		if strings.HasPrefix(container.Name, "build-step-git-source") || strings.HasPrefix(container.Name, "step-git-source") {
```
Comment (dgozalo) [inline pkg/tekton/pipeline_info.go]: Same as with the pod. The second condition is true for the first one.

File: pkg/builds/build_info.go (hunk 1 above)
Comment (abayer) [reply pkg/builds/build_info.go]: I thought `HasPrefix` looked for, in regex terms, `/^build-step-git-source.*/`?

File: pkg/logs/tekton_logging.go (hunk 2 above)
Comment (abayer) [reply pkg/logs/tekton_logging.go]: @hferentschik did this last week. =)

File: pkg/builds/build_info.go (hunk 1 above)
Comment (dgozalo) [reply pkg/builds/build_info.go]: Nvm me! it's HasPrefix and not Contains :D :D :D

Comment (abayer) [reply pkg/builds/build_info.go]: =) I actually considered going with `Contains` but decided to make sure we didn't get confused by someone actually naming a step something like `Step Git Source`, which would end up as a container named `step-step-git-source`, and wouldn't be the right container to look at, etc, etc...

File: pkg/logs/tekton_logging.go (hunk 2 above)
Comment (hferentschik) [reply pkg/logs/tekton_logging.go]: Now one can find all resources related to a pipeline execution (including the pipeline activity) using the same label, something like:

```
for i in pipelineactivity pipeline pipelinestructure taskrun pipelinerun task pod; do 
   kubectl get "$i" --show-kind --no-headers --ignore-not-found -l repository=demo,build=<x>
  done
```

**Comment by hferentschik** [comment]
@abayer looks good to me. Does merging this require syncing updates of the Tekton controller Helm charts as well?

**Pull request merged**